enterprise/api/v2/views.py

```diff
@@ -0,0 +1,97 @@
+from enterprise.constants import COURSE_KEY_URL_PATTERN
+
+LOGGER = getLogger(__name__)
+
+
+class EnterpriseCustomerViewSetV2(EnterpriseCustomerViewSet):
+    """
+    API views for the ``enterprise-customer`` API endpoint.
+    """
+
+    queryset = models.EnterpriseCustomer.active_customers.all()
+    serializer_class = serializers.EnterpriseCustomerSerializer
+
+    USER_ID_FILTER = 'enterprise_customer_users__user_id'
+    FIELDS = (
+        'uuid', 'slug', 'name', 'catalog', 'active', 'site', 'enable_data_sharing_consent',
+        'enforce_data_sharing_consent',
+    )
+    filter_fields = FIELDS
+    ordering_fields = FIELDS
+
+    @detail_route()
+    def courses(self, request, pk=None):  # pylint: disable=invalid-name,unused-argument
+        """
+        Retrieve the list of courses contained within the enterprise catalogs linked to this enterprise.
+
+        Only courses with active course runs are returned. A course run is considered active if it is currently
+        open for enrollment, or will open in the future.
+        """
+        enterprise_customer = self.get_object()
+        combined_content_filter = {}
+        for catalog in enterprise_customer.enterprise_customer_catalogs.all():
+            if not catalog.content_filter:
+                continue
+            update_content_filters(combined_content_filter, catalog.content_filter)
+
+        ######################### CONSTRUCTION ZONE #########################
+        # - Now that we have combined_content_filter, do something with it here
+        # - Call the discovery service using it
+        # - Get the results and serialize it
+        # (Looks like you can use something like get_catalog_courses, except while
+        #  accepting a query string. Need to figure out how to take this dict we created
+        #  and turn it into the proper datatype for the `querystring` variable)
+        catalog_api = CourseCatalogApiClient(request.user, enterprise_customer.site)
+        courses = catalog_api.get_search_results(pk, request.GET) # TO CHANGE!
+        serializer = serializers.EnterpriseCatalogCoursesReadOnlySerializer(courses)
+
+        # Add enterprise related context for the courses.
+        serializer.update_enterprise_courses(enterprise_customer, catalog_id=enterprise_customer.catalog)
+        return get_paginated_response(serializer.data, request)
+        #####################################################################
+
+def update_content_filters(combined_content_filter, new_content_filter):
+    """
+    Helper method for combining 2 content filter dicts
+    """
+    for filter_key, filter_value in new_content_filter.items():
+        if filter_key in combined_content_filter:
+            old_value = combined_content_filter[filter_key]
+            if isinstance(filter_value, list):
+                combined_content_filter[filter_key] = list(set(old_value) | set(filter_value))
+            elif filter_value != old_value:
```

Comment:
you probably already are planning to do this, but there are some keys that creating a list won't work for, in particular, `date_lte` sort of fields. just a note to account for that 😄 
i also think this should function should live in enterprise/api/utils.py